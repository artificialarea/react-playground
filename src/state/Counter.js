import React from 'react';

class Counter extends React.Component {

  state = {
    count: 0,
  }

  // General Rule:
  // When adding an event listener function to a React class, use an arrow function to make sure that this is referring to the class instance.
  handleButtonClick = () =>  {
    const newCount = this.state.count + 1;
    this.setState({
      count: newCount
    })
    console.log(this.state.count);
  }

  render() {
    return (
      <div>
        <p> The current count: {this.state.count}</p>
        <button 
          onClick={this.handleButtonClick}
        >
          +
        </button>
      </div>
    )
  }
}

export default Counter;