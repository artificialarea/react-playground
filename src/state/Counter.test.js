import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Counter from './Counter';

describe('<Counter />', () => {

  it('render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Counter />, div)
    ReactDOM.unmountComponentAtNode(div)
  });

  it('render UI without crashing', () => {
    expect(
      renderer.create(<Counter />).toJSON()
    ).toMatchSnapshot();
  });

});