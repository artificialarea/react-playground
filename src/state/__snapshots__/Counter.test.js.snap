// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> render UI without crashing 1`] = `
<div>
  <p>
     The current count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
